### PR TITLE
Reformatted GPSTracker.java and wrote documentation

### PR DESCRIPTION
**Pull request by ravifrancesco**

### CODE REVIEW

File: app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java [hunk 1]
```diff
@@ -116,17 +99,53 @@ public Location getLocation() {
...
     }
 
+    /**
+     *
+     * @return latitude (in degrees)
+     */
     public double getLatitude(){
+
+        double latitude = 0;
+
```
Comment (bgabriel998) [inline app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java:148]: `return location != null ?  location.getLatitude() : 0.0;` would be simpler (and avoid creating new variable) but maybe not as clear.

File: app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java [hunk 2]
```diff
@@ -43,71 +52,45 @@
     // Declaring a Location Manager
     protected LocationManager locationManager;
 
+    // User Point
+    UserPoint userPoint;
+
```
[the remark this answers is not in the document]
Comment (ravifrancesco) [reply app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java:57]: It shouldn't create problems, I put a reference to gpstracker in the UserPoint if we want to change something, but a part from that it is never used

File: app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java [hunk 3]
```diff
@@ -43,71 +52,45 @@
...
+     * Method used to get updates on the location. Once called it will request a new location to the
+     * location manager and update the location.
+     * Before requesting the location the method will check if the requisites are satisfied, and
+     * will request a new location via NETWORK_PROVIDER if it is available, otherwise it will request
+     * it through the GPS_PROVIDER (less precise)
+     *
+     * In case it fails to do so it will stop the program and print the stack strace of the error
+     */
+    private void getLocation() {
         try {
```
[the remark this answers is not in the document]
Comment (ravifrancesco) [reply app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java:80]: There is a catch before the end of the method

**Review by bgabriel998 — APPROVED** [review]

File: app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java (hunk 1 above)
Comment (ravifrancesco) [reply app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java:148]: You're right, I will fix that now

**Review by bgabriel998 — APPROVED** [review]

File: app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java [hunk 4]
```diff
@@ -116,80 +99,146 @@ public Location getLocation() {
...
+        //check the permission
+        if (ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(mContext, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions((Activity) mContext, new String[]{android.Manifest.permission.ACCESS_FINE_LOCATION, Manifest.permission.ACCESS_COARSE_LOCATION}, 101);
         }
+        // request location
+        locationManager.requestLocationUpdates(
+                selectedProvider,
+                MIN_TIME_BW_UPDATES,
+                MIN_DISTANCE_CHANGE_FOR_UPDATES, this);
+        // set new location
```
Comment (bgabriel998) [inline app/src/main/java/com/github/ravifrancesco/softwaredevproject/GPSTracker.java:135]: `MIN_TIME_BW_UPDATES` -> the time should be given in ms. The comment next to the definition says seconds (x1000 needed)

**Review by bgabriel998 — CHANGES_REQUESTED** [review]

**Review by giommok — APPROVED** [review]
Looks better!

**Review by bgabriel998 — COMMENTED** [review]
OK !

**Review by bgabriel998 — APPROVED** [review]
OK!

**Pull request merged**